File: src/main/java/rest/AuthResource.java
package rest;

import dao.DaoPersona;
import entity.Persona;
import org.hibernate.validator.constraints.NotEmpty;
import usecase.CorreoUseCase;
import util.JWTUtils;

import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;
import java.util.Optional;

@Path("/auth")
@Consumes(MediaType.APPLICATION_JSON)
@Produces(MediaType.APPLICATION_JSON)
public class AuthResource {

    private final DaoPersona daoPersona;

    private final JWTUtils jwtUtils;

    private final CorreoUseCase correoUseCase;

    public AuthResource(DaoPersona daoPersona, JWTUtils jwtUtils, CorreoUseCase correoUtils){
        this.daoPersona = daoPersona;
        this.jwtUtils = jwtUtils;
        this.correoUseCase = correoUtils;
    }

    @POST
    public Response credencialesCorrectas(Persona persona){
        Optional<Persona> personaIdentificada = daoPersona.darPersonaPorCredenciales(persona);
        return personaIdentificada.isPresent()?
                Response.ok(jwtUtils.darToken(personaIdentificada.get())).build() :
                Response.status(Response.Status.BAD_REQUEST).build();
    }

    @POST
    @Path("/pass")
    public Response olvidePass(@QueryParam("email") @NotEmpty String email){
        boolean envioExitoso = correoUseCase.enviarCorreo(email);
        return envioExitoso ?
                Response.ok().build() :
                Response.status(Response.Status.BAD_REQUEST).build();
    }
}

File: src/main/java/rest/AuthResource.kt
package rest

import entity.Error
import entity.Persona
import filter.ValidPersonaLogin
import org.hibernate.validator.constraints.NotEmpty
import org.jetbrains.annotations.NotNull
import usecase.CorreoUseCase
import usecase.PersonaUseCase
import javax.ws.rs.*
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("/auth")
@Consumes(MediaType.APPLICATION_JSON)
@Produces(MediaType.APPLICATION_JSON)
class AuthResource(private val personaUseCase: PersonaUseCase, private val correoUseCase: CorreoUseCase) {

    @POST
    fun credencialesCorrectas(@NotNull @ValidPersonaLogin persona: Persona): Response {
        val resultado = personaUseCase.darPersonaPorCredenciales(persona)
        return when(resultado){
            is Error -> Response.status(Response.Status.BAD_REQUEST).entity(resultado).build()
            else -> Response.ok(resultado).build()
        }
    }

    @POST
    @Path("/pass")
    fun olvidePass(@QueryParam("email") @NotEmpty email: String): Response {
        val resultado = correoUseCase.enviarCorreo(email)
        return when(resultado){
            is Error -> Response.status(Response.Status.BAD_REQUEST).entity(resultado).build()
            else -> Response.ok(resultado).build()
        }
    }
}